lib/translator/inquiry.cc

```diff
@@ -17,6 +17,79 @@
 
 namespace translator {
 
+// anonymous namepace for helper functions that are used in this file
+namespace {
+
+// The SPT field of the ExtendedInquiryData depends on the DPC field of the
+// IdentifyNamespaceData
+uint8_t GetSptValueFromDPC(
+    const nvme_defs::IdentifyNamespace &identify_namespace_data) {
+  uint8_t dpc = 0;
+  dpc |= identify_namespace_data.dpc.pit3;
+  dpc |= identify_namespace_data.dpc.pit2 << 1;
+  dpc |= identify_namespace_data.dpc.pit1 << 2;
+
+  uint8_t spt;
+  // SPT shall be translated using the DPC field of the Identify
+  // Namespace Data Structure. Refer to:
+  // https://nvmexpress.org/wp-content/uploads/NVM_Express_-_SCSI_Translation_Reference-1_5_20150624_Gold.pdf
+  // Section 6.1.5
+  switch (dpc) {
+    case 0b001:
+      spt = 0b000;
+      break;
+    case 0b010:
+      spt = 0b010;
+      break;
+    case 0b011:
+      spt = 0b001;
+      break;
+    case 0b100:
+      spt = 0b100;
+      break;
+    case 0b101:
+      spt = 0b011;
+      break;
+    case 0b110:
+      spt = 0b101;
+      break;
+    case 0b111:
+      spt = 0b111;
+      break;
+  }
+  return spt;
+}
+}  // anonymous namespace
+
+// Creates and validates a Inquiry Command struct
+StatusCode RawToScsiCommand(absl::Span<const uint8_t> raw_cmd,
```

Comment:
This function can be removed. Almost all of this validation, spare validating the size of the scsi span, is provided upstream of translate inquiry